### PR TITLE
Sw sys admin add events

### PR DESCRIPTION
**Pull request by SWvheerden**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (CjS77) [inline .vscode/launch.json]: Please add `.vscode` folder to .gitignore and repush

File: Cargo.toml [hunk 2]
```diff
@@ -12,12 +12,16 @@ actix = "0.5"
 actix-web = "0.6"
 diesel = { version = "1.3.0", features = ["r2d2"] }
 dotenv = "0.10"
-bigneon_db = { git="https://github.com/big-neon/bn-db.git" }
+# bigneon_db = { git="https://github.com/big-neon/bn-db.git" }
+bigneon_db = { path = "../bn-db" }
```
Comment (CjS77) [inline Cargo.toml]: Tests won't pass with this like this :( If versioning is an issue, I'm pretty sure you can reference a specific
commit, even if it's still a PR, so
`bigneon_db = { git="https://github.com/big-neon/bn-db.git#a3457b87fcd" }`
or whatever

File: .travis.yml [hunk 3]
```diff
@@ -20,7 +20,10 @@ env:
   TEST_DATABASE_ADMIN_URL: postgres://postgres:password123@localhost/bigneon_test
   TOKEN_SECRET: travis_secret
   TOKEN_ISSUER: bg-on-travis
+<<<<<<< HEAD
+=======
   COOKIE_SECRET_KEY: 8373b8987dab284b39c93bcda294bc294b284b72b29cd9028bca
+>>>>>>> 89d6a58c666373df329513f8b42a1726b7df35c1
```
Comment (mikethetike) [inline .travis.yml]: Looks like some merge stuff in here

File: src/controllers/login.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
+use actix_web::error;
+use actix_web::{Error, HttpRequest, HttpResponse, Json, Responder, State};
+use bigneon_db::models::User;
+use crypto::sha2::Sha256;
+use jwt::{Header, Registered, Token};
+use serde_json;
+use server::AppState;
+
+#[derive(Serialize, Deserialize)]
```
Comment (mikethetike) [inline src/controllers/login.rs]: I have renamed this file to a different file... it should actually be deleted

File: src/controllers/mod.rs [hunk 5]
```diff
@@ -1,3 +1,7 @@
 pub mod artists;
 pub mod auth;
+pub mod events;
+pub mod login;
```
Comment (mikethetike) [inline src/controllers/mod.rs]: This file has changed to `auth` so you should be able to remove it from here

File: src/routing.rs [hunk 6]
```diff
@@ -21,7 +23,23 @@ pub fn routes(config: &Config, app: App<AppState>) -> App<AppState> {
             r.method(Method::GET).with(artists::index);
             r.method(Method::POST).with(artists::create);
         })
-        .route("/auth/token", Method::POST, auth::token)
-        .route("/login", Method::POST, sessions::create)
-        .route("/logout", Method::DELETE, sessions::destroy)
+        //.route("/auth/token", Method::POST, auth::token)
+        //.route("/login", Method::POST, sessions::create)
+        //.route("/logout", Method::DELETE, sessions::destroy)
+        .route("/venues", Method::GET, venues::index)
```
Comment (mikethetike) [inline src/routing.rs]: Switch to `.resource` and add middleware

File: tests/support/util.rs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+use std::str;
+
+pub fn unwrap_body_to_string(response: &HttpResponse) -> Result<&str, &'static str> {
+    match response.body() {
+        Binary(binary) => Ok(str::from_utf8(binary.as_ref()).unwrap()),
+        _ => Err("Unexpected response body"),
+    }
+}
+
+pub fn create_test_request(database: Box<ConnectionGranting>) -> TestRequest<AppState> {
```
Comment (mikethetike) [inline tests/support/util.rs]: I think this has also been removed....

File: tests/unit/login.rs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+use jwt::Header;
+use jwt::Registered;
+use jwt::Token;
+use serde_json;
+use support::database::TestDatabase;
+use support::util;
+use support::util::create_test_request;
+
+#[test]
+fn get_token_valid_data() {
```
Comment (mikethetike) [inline tests/unit/login.rs]: I think this file has also been renamed/moved

**Pull request merged**